### PR TITLE
fix: Adds a deprecation warning for SEND_BROKEN_LINK_EMAILS

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

The `SEND_BROKEN_LINK_EMAILS` setting has been removed since Django 1.8. This PR adds a deprecation warning to the v3 base. 

## Related resources

* #7419 

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/utils/setup.py [hunk 1]
```diff
@@ -33,6 +35,14 @@ def validate_settings():
         raise ImproperlyConfigured("django CMS requires django.template.context_processors.request in "
                                    "'django.template.backends.django.DjangoTemplates' context processors.")
 
+    if (
+        hasattr(settings, "SEND_BROKEN_LINK_EMAILS") and  # noqa: W504
+        "django.middleware.common.BrokenLinkEmailsMiddleware" not in getattr(settings, "MIDDLEWARE", [])
+    ):
+        warnings.warn('The setting "SEND_BROKEN_LINK_EMAILS" will not be honored by django CMS as of version 4.1. '
```
Comment (vinitkumar) [inline cms/utils/setup.py:42]: Similar to the discussion here, let's add `stacklevel=2` to the `warnings.warn` method, to make it easier for the end users of this to spot where they are using this in their code.

https://github.com/django-cms/django-cms/pull/7410#discussion_r1006064988

**Review by vinitkumar — COMMENTED** [review]
Please look at the my comment.

File: cms/utils/setup.py (hunk 1 above)
Comment (fsbraun) [reply cms/utils/setup.py:42]: Hi @vinitkumar ! Thanks, but the `SEND_BROKEN_LINK_EMAIL` is a setting which always resides in the settings file and I will not find on the stack. Hence, for this case, the stack level does not help. 

But the `stacklevel=2` is great stuff. Thank you so much. Before reading your other comment I was not aware of it.

**Comment by fsbraun** [comment]
Yeah, just replied. 😎. If you're ok, would you resolve the conversation?

**Comment by vinitkumar** [comment]
@fsbraun Just resolved my conversation. Please resolve the conflicts in CHANGELOG.rst and we are good to go here.

**Review by vinitkumar — APPROVED** [review]

**Comment by fsbraun** [comment]
Thanks, @vinitkumar !!

**Pull request merged**